core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -245,8 +245,7 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
   }
 
   test("cannot call addFile with different paths that have the same filename") {
```

Comment:
Hi, @heary-cao . This PR is very useful. Could you replace more `val dir = Utils.createTempDir()` with new `withTempDir` in this file? For example, [the above test case at line 235](https://github.com/apache/spark/pull/23151/files#diff-8d5858d578a2dda1a2edb0d8cefa4f24R235).

Reply:
ok, i have update it. please review it. thanks.